### ISSUE TITLE
Libre 3 protocol

### ISSUE THREAD

**Issue by tonnyleonard** [glucometers-tech/glucometer-protocols#33]
Did anyone attempt to read a Libre 3 sensor?
I do have the sensor and the official app, but so far the sensor is not applied, although I paired it already. I intend to use it with a diabetic dog, but I understood that the data - at least with Libre 2 - was sometimes crappy, that is, there were wrong readings - therefore I would like to get the raw measurements and process the data on my own.
Originally I intended to use a Raspi 3 (it has BLE incorporated), but then I discovered that they decided to encrypt the communication...
I will test it with the Libre 2 scripts, but if it doesn't work, I guess I will have to do some reverse engineering on my own...

**Comment by gui-dos** [comment]
@tonnyleonard:

Yes, the protocols have been changed substantially: Libre 3 now uses the ECDH key agreement protocol and the AES 128 CCM block cipher.

I summarize in the single https://github.com/gui-dos/DiaBLE/blob/main/DiaBLE/Libre3.swift my attempts to reverse the protocols.

**Comment by isotherm** [comment]
The protocol is the same, but the format of the some of the messages has changed (comments and some other values are removed). I have a pull request adding support to glucometerutils at https://github.com/glucometers-tech/glucometerutils/pull/131. It also contains some sample data output and could be used to produce the appropriate documentation here.

**Comment by ryan-carpenter** [comment]
> Did anyone attempt to read a Libre 3 sensor?

I successfully used the the pull request at https://github.com/glucometers-tech/glucometerutils/pull/131 to dump data from a Freestyle Libre 3 Reader.
